src/dynimage.rs

```diff
@@ -761,13 +761,31 @@ fn open_impl(path: &Path) -> ImageResult<DynamicImage> {
     load(fin, format)
 }
 
+/// Open the image located at the path specified.
+/// The image's format is determined from the file header magic bytes.
+pub fn open_magic<P>(path: P) -> ImageResult<DynamicImage>
+where
+    P: AsRef<Path>,
+{
+    // thin wrapper function to strip generics before calling open_magic_impl
+    open_magic_impl(path.as_ref())
+}
+
+fn open_magic_impl(path: &Path) -> ImageResult<DynamicImage> {
```

Comment:
What purpose does`open_magic_impl` serve, can't the body of this function be used directly in `open_magic`?

I also wonder if we should use magic bytes as primary way of detecting file type in `open`, and use a file-extension fall-back if magic byte detection failed. This should be more versatile than using file-extensions.

Reply:
open_magic_impl() is just a pattern I copied from how [open() was implemented](https://github.com/image-rs/image/blob/master/src/dynimage.rs#L723).
Indeed code can be used directly in open_magic(), thanks for noticing!

I think magic bytes detection would be a good default, but should I try address that in this PR or a later one?

Reply:
I don't see why you shouldn't do it in this PR, we could simply wait a few days before merging to see if anyone has any objections.

Reply:
Okay, sure